### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by cwbit**
clarify purpose of module after the discussion today in IRC

### CODE REVIEW

**Comment by markstory** [comment]
Seems reasonable enough to me, but this is @jadb 's baby.

File: README.md [hunk 1]
```diff
@@ -4,7 +4,16 @@ CakePHP 3 Codeception Module
...
+***What?*** A [codeception](http://codeception.com) module to test your CakePHP 3 powered application.
+
+***Why?*** Using Codeception with CakePHP opens up a whole new set of testing capabilities for your app letting you do
+
+- **front-end testing** _(i.e. browser-based workflow tests)_
+  - [x] **[[Functional Tests]](http://codeception.com/docs/05-FunctionalTests)**  - does our Cake App run the way we expect it to
+  - [x] **[[Acceptance Tests]](http://codeception.com/docs/04-AcceptanceTests)**  - does our Cake App do what the customer/user expects it to
+- **back-end testing** _(i.e. direct app function calls)_
+   - [x] **[[API Tests]](http://codeception.com/docs/10-WebServices)** - does my Cake App's API work 
+   - [ ] **[[Unit Tests]](http://codeception.com/docs/06-UnitTests)** - do our App's internal functions do what they should
```
Comment (jadb) [inline README.md]: While I like how it gives more details right off the bat, I personally don't really like the 'marketing' tone it has.

I suggest instead something like:

```
A [codeception](http://codeception.com) module to test your CakePHP 3 
powered application. Codeception opens up a whole new set of testing 
capabilities:
```

And for the list, I wouldn't split (front/back), would drop the checkboxes and styles. I would also follow this template:

```
- [Acceptance Tests]() - real browser tests to cover standard but complex 
scenarios from a user's perspective.
```

Comment (ceeram) [inline README.md]: why is this one not marked with an X like the others?

Comment (cwbit) [inline README.md]: because it is available without Codeception thru regular PHPUnit tests

Comment (jadb) [inline README.md]: @cwbit that is true but using this module adds even more unit testing capabilities to the already solid cake test suite and most importantly, unifies the API for writing tests (across all suites), if you're writing `Cept` or `Cest` unit test cases and/or extending the `Codeception\TestCase`.

**Comment by cwbit** [comment]
regarding the front/back split. someone who is looking at this stuff for the first time will be confused what the difference is. I know, because I was. As soon as you understand that Functional Testing is the front-end equivalent of back-end Unit tests, things start to make more sense. I would keep the split.

**Comment by jadb** [comment]
@cwbit if you think the split helps, keep it.

**Comment by jadb** [comment]
Thanks!

**Pull request merged**